src/main/java/org/folio/consortia/service/impl/SyncPrimaryAffiliationServiceImpl.java

```diff
@@ -81,44 +68,30 @@ private SyncPrimaryAffiliationBody buildSyncPrimaryAffiliationBody(String tenant
 
-    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentTenantContext))) {
-      TenantEntity tenantEntity = tenantService.getByTenantId(tenantId);
-      IntStream.range(0, userList.size())
-        .sequential()
-        .forEach(idx -> {
-          var user = userList.get(idx);
-          log.info("Processing users: {} of {}", idx + 1, userList.size());
+    TenantEntity tenantEntity = tenantService.getByTenantId(tenantId);
+    IntStream.range(0, userList.size()).sequential().forEach(idx -> {
+      var user = userList.get(idx);
+      log.info("Processing users: {} of {}", idx + 1, userList.size());
+      Page<UserTenantEntity> userTenantPage = userTenantRepository.findByUserId(UUID.fromString(user.getId()), PageRequest.of(0, 1));
 
-          // context changes in every iteration and folioExecutionContext become an empty, so we should set saved context again.
-          try (var context2 = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentTenantContext))) {
-            Page<UserTenantEntity> userTenantPage = userTenantRepository.findByUserId(UUID.fromString(user.getId()), PageRequest.of(0, 1));
-            if (userTenantPage.getTotalElements() > 0) {
-              log.info("Primary affiliation already exists for tenant/user: {}/{}", tenantId, user.getUsername());
-            } else {
-              userTenantService.createPrimaryUserTenantAffiliation(consortiumId, tenantEntity, user.getId(), user.getUsername());
-              if (ObjectUtils.notEqual(centralTenantId, tenantEntity.getId())) {
-                userTenantService.save(consortiumId, createUserTenant(centralTenantId, user), true);
-              }
-              // context changes in userTenantService.save(), so we should set saved context again.
-              try (var context3 = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentTenantContext))) {
-                sendCreatePrimaryAffiliationEvent(tenantEntity, user);
-              }
-            }
-          }
+      if (userTenantPage.getTotalElements() > 0) {
+        log.info("Primary affiliation already exists for tenant/user: {}/{}", tenantId, user.getUsername());
+      } else {
+        userTenantService.createPrimaryUserTenantAffiliation(consortiumId, tenantEntity, user.getId(), user.getUsername());
+        if (ObjectUtils.notEqual(centralTenantId, tenantEntity.getId())) {
+          userTenantService.save(consortiumId, createUserTenant(centralTenantId, user), true);
+        }
+        sendCreatePrimaryAffiliationEvent(tenantEntity, user);
+      }
 
-      });
-      log.info("Successfully created primary affiliations for tenant {}", tenantId);
-    } catch (Exception e) {
-      log.error("Failed to create primary affiliations for tenant {}", tenantId, e);
```

Comment:
I think we need teturn this catch, in order if some particular user was not migrated - entire operation should not be failed

Reply:
reverted try catch